yarn-project/circuits.js/src/structs/kernel/public_kernel_circuit_public_inputs.ts

```diff
@@ -84,6 +94,68 @@ export class PublicKernelCircuitPublicInputs {
     return !this.publicTeardownCallStack[0].isEmpty();
   }
 
+  recombineAccumulatedData(): CombinedAccumulatedData {
+    const newNoteHashes: Tuple<Fr, typeof MAX_NEW_NOTE_HASHES_PER_TX> = mapTuple(
+      mergeAccumulatedData(this.endNonRevertibleData.newNoteHashes, this.end.newNoteHashes, MAX_NEW_NOTE_HASHES_PER_TX),
+      n => n.value,
+    );
+
+    const newNullifiers: Tuple<Fr, typeof MAX_NEW_NOTE_HASHES_PER_TX> = mapTuple(
+      mergeAccumulatedData(this.endNonRevertibleData.newNullifiers, this.end.newNullifiers, MAX_NEW_NOTE_HASHES_PER_TX),
+      n => n.value,
+    );
+
+    const newL2ToL1Msgs = mergeAccumulatedData(
+      this.endNonRevertibleData.newL2ToL1Msgs,
+      this.end.newL2ToL1Msgs,
+      MAX_NEW_L2_TO_L1_MSGS_PER_TX,
+    );
+
+    const noteEncryptedLogHashes = mergeAccumulatedData(
+      this.endNonRevertibleData.noteEncryptedLogsHashes,
+      this.end.noteEncryptedLogsHashes,
+      MAX_NOTE_ENCRYPTED_LOGS_PER_TX,
+    );
+    const noteEncryptedLogPreimagesLength = new Fr(arrayNonEmptyLength(noteEncryptedLogHashes, x => x.isEmpty()));
+
+    const encryptedLogHashes = mergeAccumulatedData(
+      this.endNonRevertibleData.encryptedLogsHashes,
+      this.end.encryptedLogsHashes,
+      MAX_ENCRYPTED_LOGS_PER_TX,
+    );
+    const encryptedLogPreimagesLength = new Fr(arrayNonEmptyLength(encryptedLogHashes, x => x.isEmpty()));
+
+    const unencryptedLogHashes = mergeAccumulatedData(
+      this.endNonRevertibleData.unencryptedLogsHashes,
+      this.end.unencryptedLogsHashes,
+      MAX_ENCRYPTED_LOGS_PER_TX,
+    );
+
+    const unencryptedLogPreimagesLength = new Fr(arrayNonEmptyLength(unencryptedLogHashes, x => x.isEmpty()));
+
+    const publicDataUpdateRequests = mergeAccumulatedData(
+      this.endNonRevertibleData.publicDataUpdateRequests,
+      this.end.publicDataUpdateRequests,
+      MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX,
+    );
+
+    const gasUsed = this.endNonRevertibleData.gasUsed.add(this.end.gasUsed);
+
+    return CombinedAccumulatedData.from({
+      newNoteHashes,
+      newNullifiers,
+      newL2ToL1Msgs,
+      noteEncryptedLogsHash: noteEncryptedLogHashes[0].value,
```

Comment:
NOTE, this is obviously wrong, but we're only presently using this recombined value to get the sorted public data update requests. I will be dealing with the log hashes immediately.